portfolio/src/main/java/com/google/sps/servlets/CreatePuzzleServlet.java

```diff
@@ -64,14 +64,18 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
     Puzzle puzzle = new Puzzle();
 
     puzzle.setImageUrl(uploadedFileUrl);
+    puzzle.setDifficulty(difficulty);
 
     PuzzleDao dao = new PuzzleDao();
 
     //Need to double check that puzzle is passed by reference
     dao.create(puzzle);
 
+    Long puzzleId = puzzle.getPuzzleId();
+    String redirectURL = "/ViewPuzzle?puzzleid=%s" + String.valueOf(puzzleId);
+    
     // Send the user to view puzzle page.
-    response.sendRedirect("/ViewPuzzle"); 
+    response.sendRedirect(redirectURL); 
```

Comment:
I took a look at @markcalvario code in #13 and it is expecting a response and not a redirect.  Have this return the puzzle as JSON instead: 
```suggestion
    response.setContentType("application/json;");
    response.getWriter().println(new Gson().toJson(puzzle));
```